AVX2 intrinsics unit test fails to compile

Compiling on a Linux system with a i5-6600K CPU (avx2 enabled) with gcc 7.1.1 fails to build the intrinsics unit test with the error below.

The Travis CI build is also failing.

Everything else builds and runs successfully: it is just this unit test that seems to be the problem.

```
[ 40%] Building CXX object tests/unit/CMakeFiles/test.exe.dir/test_intrin.cpp.o
In file included from /home/bcumming/software/github/nestmc-proto/tests/unit/test_intrin.cpp:3:0:
/home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp: In function ‘__m256d nest::mc::multicore::nmc_mm256_exp_pd(__m256d)’:
/home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp:152:35: warning: AVX vector return without AVX enabled changes the ABI [-Wpsabi]
 __m256d nmc_mm256_exp_pd(__m256d x) {
                                   ^
/home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp: In function ‘void nest::mc::multicore::nmc_mm256_print_pd(__m256d, const char*)’:
/home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp:76:6: note: The ABI for passing parameters with 32-byte alignment has changed in GCC 4.6
 void nmc_mm256_print_pd(__m256d x, const char *name) {
      ^~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/immintrin.h:41:0,
                 from /home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp:10,
                 from /home/bcumming/software/github/nestmc-proto/tests/unit/test_intrin.cpp:3:
/usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/avxintrin.h: In function ‘__m256d nest::mc::multicore::nmc_mm256_exp_pd(__m256d)’:
/usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/avxintrin.h:218:1: error: inlining failed in call to always_inline ‘__m256d _mm256_blendv_pd(__m256d, __m256d, __m256d)’: target specific option mismatch
 _mm256_blendv_pd (__m256d __X, __m256d __Y, __m256d __M)
 ^~~~~~~~~~~~~~~~
In file included from /home/bcumming/software/github/nestmc-proto/tests/unit/test_intrin.cpp:3:0:
/home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp:217:7: note: called from here
     x = _mm256_blendv_pd(x, detail::nmc_m256d_nan, is_nan);
     ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/immintrin.h:41:0,
                 from /home/bcumming/software/github/nestmc-proto/src/backends/multicore/intrin.hpp:10,
                 from /home/bcumming/software/github/nestmc-proto/tests/unit/test_intrin.cpp:3:
/usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/avxintrin.h:218:1: error: inlining failed in call to always_inline ‘__m256d _mm256_blendv_pd(__m256d, __m256d, __m256d)’: target specific option mismatch
 _mm256_blendv_pd (__m256d __X, __m256d __Y, __m256d __M)

```

--- a/tests/unit/CMakeLists.txt
+++ b/tests/unit/CMakeLists.txt
@@ -47,7 +47,6 @@ set(TEST_SOURCES
     test_event_binner.cpp
     test_filter.cpp
     test_fvm_multi.cpp
-    #test_intrin.cpp
     test_mc_cell_group.cpp
     test_lexcmp.cpp
     test_mask_stream.cpp
@@ -87,6 +86,9 @@ set(TEST_SOURCES
     stats.cpp
 )
 
+if(NMC_VECTORIZE_TARGET STREQUAL "AVX2")
+    list(APPEND TEST_SOURCES test_intrin.cpp)
+endif()
 
 set(TARGETS test.exe)
 

